apps/bundle-analyzer/components/ui/multi-select.tsx

```diff
@@ -0,0 +1,184 @@
+'use client'
+
+import * as React from 'react'
+import { Button } from '@/components/ui/button'
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from '@/components/ui/popover'
+import { ChevronDown, CheckIcon } from 'lucide-react'
+import { cn } from '@/lib/utils'
+
+export interface MultiSelectPropOption {
+  value: string
+  label: string
+  icon?: React.ReactNode
+}
+
+interface MultiSelectProps {
+  options: MultiSelectPropOption[]
+  value: string[]
+  onValueChange: (value: string[]) => void
+  placeholder?: string
+  emptyMessage?: string
+  selectionName?: string
+  className?: string
+  triggerClassName?: string
+  triggerIcon?: React.ReactNode
+  'aria-label'?: string
+  size?: 'sm' | 'default'
+}
+
+export function MultiSelect({
+  options,
+  value,
+  onValueChange,
+  placeholder = 'Select items...',
+  selectionName = 'item',
+  className,
+  triggerClassName,
+  triggerIcon,
+  'aria-label': ariaLabel,
+}: MultiSelectProps) {
+  const [open, setOpen] = React.useState(false)
+
+  function handleToggle(optionValue: string, checked: boolean) {
+    const newValue = checked
+      ? [...value, optionValue]
+      : value.filter((v) => v !== optionValue)
+    if (newValue.length > 0) {
+      onValueChange(newValue)
+    }
+  }
+
+  // TODO: more robust pluralization
```

Comment:
```suggestion
  // TODO: more robust pluralization. This fails for `selectionName` values like octopus, fairy, mouse, and child
```

The solution would be to accept the plural form as a parameter

Reply:
> The solution would be to accept the plural form as a parameter

great idea, updated